nexus/client.go

```diff
@@ -402,43 +409,49 @@ func failureFromResponse(response *http.Response, body []byte) (Failure, error)
 	return failure, err
 }
 
-func failureFromResponseOrDefault(response *http.Response, body []byte, defaultMessage string) Failure {
-	failure, err := failureFromResponse(response, body)
+func (c *HTTPClient) failureFromResponseOrDefault(response *http.Response, body []byte, defaultMessage string) Failure {
+	failure, err := c.failureFromResponse(response, body)
 	if err != nil {
 		failure.Message = defaultMessage
 	}
 	return failure
 }
 
-func bestEffortHandlerErrorFromResponse(response *http.Response, body []byte) error {
+func (c *HTTPClient) failureErrorFromResponseOrDefault(response *http.Response, body []byte, defaultMessage string) error {
+	failure := c.failureFromResponseOrDefault(response, body, defaultMessage)
+	failureErr := c.options.FailureConverter.FailureToError(failure)
+	return failureErr
+}
+
+func (c *HTTPClient) bestEffortHandlerErrorFromResponse(response *http.Response, body []byte) error {
 	switch response.StatusCode {
 	case http.StatusBadRequest:
-		failure := failureFromResponseOrDefault(response, body, "bad request")
-		return &HandlerError{Type: HandlerErrorTypeBadRequest, Failure: &failure}
+		failureErr := c.failureErrorFromResponseOrDefault(response, body, "bad request")
+		return &HandlerError{Type: HandlerErrorTypeBadRequest, Cause: failureErr}
 	case http.StatusUnauthorized:
-		failure := failureFromResponseOrDefault(response, body, "unauthenticated")
-		return &HandlerError{Type: HandlerErrorTypeUnauthenticated, Failure: &failure}
+		failureErr := c.failureErrorFromResponseOrDefault(response, body, "unauthenticated")
```

Comment:
Is it intentional that `StatusUnauthorized` maps to `unauthenticated`, naively I would think the mapping would be `StatusUnauthorized` -> `unauthorized`?

Reply:
We map forbidden to unauthorized. I stole it from gRPC:

https://github.com/grpc-ecosystem/grpc-gateway/blob/a7cf811e6ffabeaddcfb4ff65602c12671ff326e/runtime/errors.go#L52C1-L55C33